http-netty/src/main/java/io/micronaut/http/netty/NettyHttpHeaders.java

```diff
@@ -141,16 +156,34 @@ public Optional<String> findFirst(CharSequence name) {
     public MutableHttpHeaders add(CharSequence header, CharSequence value) {
         validateHeader(header, value);
         nettyHeaders.add(header, value);
+        onModify(header);
         return this;
     }
 
     @Override
     public MutableHeaders set(CharSequence header, CharSequence value) {
         validateHeader(header, value);
         nettyHeaders.set(header, value);
+        onModify(header);
         return this;
     }
 
+    private void onModify(CharSequence header) {
+        if (contentType != null && header.equals(HttpHeaders.CONTENT_TYPE)) {
```

Comment:
can't use equals here, it's CharSequence. Is this called on the hot path? maybe it's fine to just unset all the caches and be done with it

Reply:
Removed, looks like the the modifications are mostly for the response anyway